v2/extensions/dataref_extension.go

```diff
@@ -0,0 +1,36 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package extensions
+
+import (
+	"github.com/cloudevents/sdk-go/v2/event"
+	"net/url"
+)
+
+const DataRefExtensionKey = "dataref"
+
+type DataRefExtension struct {
+	DataRef string `json:"dataref"`
+}
+
+func AddDataRefExtension(e *event.Event, dataRef string) error {
+	if _, err := url.Parse(dataRef); err != nil {
+		return err
+	}
+	e.SetExtension(DataRefExtensionKey, dataRef)
+	return nil
+}
+
+func GetDataRefExtension(e event.Event) (*DataRefExtension, bool) {
+	if dataRefValue, ok := e.Extensions()[DataRefExtensionKey]; ok {
+		dataRefStr, ok := dataRefValue.(string)
+		if !ok {
+			return nil, false
+		}
+		return &DataRefExtension{DataRef: dataRefStr}, true
+	}
+	return nil, false
```

Comment:
Sorry Doug for this nit, but let's keep the value semantics here because using pointer can lead to (ownership) surprises (without checking the rest of the code, I assume we mostly use value semantics for core CE fields).

Reply:
Typically (convention?), if an error is returned, the value should not be used (we can add a comment to make it even more explicit). Alternatively, If needed, users can check for the zero value with https://pkg.go.dev/reflect#Value.IsZero.

Reply:
this is actually one of the reasons I prefer to return `nil` - I find the entire Zero value stuff annoying when `nil` is much more clear .  If we return a Zero struct then it's too easy for someone to think the dataref is set but with "" - which is a valid URL. But I see at least one of the other extensions does it without a pointer so I'll just hold my nose and switch it back.

Reply:
IIRC, Go is moving to making it easier to understand zero values (recently they did it with `time` for example) because `nil` has so many edges, most importantly surprising side effects for SDK users due to ownership/pointer semantics.

nit: we should add comments to these new public methods.

Reply:
done